src/pip/_internal/metadata/importlib/_dists.py

```diff
@@ -165,9 +166,14 @@ def canonical_name(self) -> NormalizedName:
 
     @property
     def version(self) -> Version:
-        if version := parse_name_and_version_from_info_directory(self._dist)[1]:
+        try:
+            version = (
+                parse_name_and_version_from_info_directory(self._dist)[1]
+                or self._dist.version
+            )
             return parse_version(version)
-        return parse_version(self._dist.version)
+        except TypeError:
+            raise BadMetadata(self._dist, reason="invalid metadata entry `version`")
```

Comment:
The change to this file seems perfectly OK to me. It targets the exception reported by the OP in #13443, and translates it to a sensible error message.